core/client/kt/verify.go

```diff
@@ -108,9 +109,9 @@ func (v *Verifier) VerifyGetEntryResponse(ctx context.Context, userID string, in
 	}
 	Vlog.Printf("✓ Sparse tree proof verified.")
 
-	sig := in.GetSmr().Signature
-	in.GetSmr().Signature = nil // Remove the signature from the object to be verified.
-	if err := v.sig.Verify(in.GetSmr(), sig); err != nil {
+	smr := *in.GetSmr()
```

Comment:
I'll add clear comments on the caller side. 
@quannguyentm Would love your input on this API design.